### PR TITLE
fixed reading lists and none with config

### LINKED ISSUE #245: fix reading several python objects via readconfig method
TODO
- [x] make the evaluation of python objects optional, possible to skip sections such as `[setup_config]`. In addition, remove the extra line that takes care of this in `cli.cli_utils.parse_config` for consistency between cli and python usage.
- [x] fix reading list (this breaks for `abs_path=True`)
- [x] fix reading None

relates to https://github.com/Deltares/hydromt_wflow/issues/116, https://github.com/Deltares/hydromt_wflow/pull/123

### PR DESCRIPTION
**Pull request by DirkEilander**
fix #245

### CODE REVIEW

**Comment by DirkEilander** [comment]
@hboisgon This small PR is ready for review. It fixes also a hydromt_wflow issue, see linked issue.

File: hydromt/config.py [hunk 1]
```diff
@@ -59,23 +66,28 @@ def configread(
...
-        sdict = dict()
+        # evaluate ini items to parse to python default objects:
+        if skip_eval or section in skip_eval_sections:
+            cfdict[section].update(dict(cf.items(section)))
+            continue  # do not evaluate
+        # numbers, tuples, lists, dicts, sets, booleans, and None
         for key, value in cf.items(section):
             try:
-                v = literal_eval(value)
-                assert not isinstance(v, tuple)  #  prevent tuples from being parsed
```
[the remark this answers is not in the document]
Comment (DirkEilander) [reply hydromt/config.py:66]: Perhaps good to double check the test for the different plugins for this.

**Comment by DirkEilander** [comment]
@hboisgon I think this one is ready to be reviewed. Could you have a look at it? I don't think it will take long.

File: hydromt/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hboisgon) [reply hydromt/config.py:66]: Also no idea what we wouldn't want to parse tuples... I also don't recall seeing tuples in any of the examples ini files

**Review by hboisgon — APPROVED** [review]

**Pull request merged**